tests/neg/RefinedProp.hs

```diff
@@ -0,0 +1,16 @@
+{-# Language GADTs #-}
+
+{-@ LIQUID "--expect-any-error" @-}
+
+module RefinedProp where
+
+import Language.Haskell.Liquid.ProofCombinators
+
+data Id where
+{-@ MkId :: Prop (Id 12) @-}
+    MkId :: Id
+data ID = Id Int
+
+-- Should error as False is supposed to not be satisfied
+{-@ bad :: { v:Prop (Id 12) | False } @-}
+bad = MkId
```

Comment:
How about using `{-@ fail bad @-}` instead of allowing any error?
```suggestion
module RefinedProp where

import Language.Haskell.Liquid.ProofCombinators

data Id where
{-@ MkId :: Prop (Id 12) @-}
    MkId :: Id
data ID = Id Int

-- Should error as False is supposed to not be satisfied
{-@ fail bad @-}
{-@ bad :: { v:Prop (Id 12) | False } @-}
bad = MkId
```

Reply:
Did not know you could do that